src/main.js

```diff
@@ -91,17 +91,9 @@ export class AwsClient {
       input = url
     }
     const signer = new AwsV4Signer(Object.assign({ url: input }, init, this, init && init.aws))
-    const signed = Object.assign({}, init, await signer.sign())
+    const signed = Object.assign(signer.body instanceof ReadableStream ? { duplex: 'half' } : {}, init, await signer.sign())
```

Comment:
potentially we could always include `{ duplex: 'half' }`, I'm not sure how necessary it is to actually run the `signer.body instanceof ReadableStream` check.